website/integrations: Slack: cleanup (#23336)
goauthentik/authentik: website/integrations/chat-communication-collaboration/slack/index.mdx

---
title: Integrate with Slack
sidebar_label: Slack
support_level: authentik
---

import SAMLProvider20265Warning from "../../_saml-provider-2026-5-warning.mdx";

## What is Slack?

> Slack is a collaboration platform for messaging, calls, file sharing, and workflow automation.
>
> -- https://slack.com

## Preparation

The following placeholders are used in this guide:

- `company.slack.com` is the FQDN of your Slack workspace.
- `authentik.company` is the FQDN of the authentik installation.

:::info
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

:::info Slack requirements
SAML SSO requires a Slack plan where custom SAML is available and a Slack Workspace Owner or Org Owner account. Optional SCIM provisioning requires a Slack Business+ or Enterprise plan and an owner or admin account that can create a Slack app with the `admin` OAuth scope.
:::

## authentik configuration

> To support the integration of Slack with authentik, create SAML property mappings and an application/provider pair. If you also want to provision Slack users and user groups from authentik, generate a Slack SCIM token in the Slack configuration section, then create a SCIM provider and assign it as a backchannel provider for the same application.

Reviewer: ```suggestion
To support the integration of Slack with authentik, you need to create SAML property mappings and an application/provider pair. If you also want to provision Slack users and user groups from authentik, generate a Slack SCIM token in the Slack configuration section, then create a SCIM provider and assign it as a backchannel provider for the same application.
```

### Create property mappings

> Slack requires the SAML assertion to include a `User.Email` attribute. You can optionally send username and name attributes.

Reviewer: ```suggestion
Slack requires the SAML assertion to include a `User.Email` attribute. You can also optionally send username and name attributes.
```

1. Log in to authentik as an administrator and open the authentik Admin interface.

> 2. Navigate to **Customization** > **Property Mappings** and click **Create**. Create the following **SAML Provider Property Mapping** entry:

Reviewer: ```suggestion
2. Navigate to **Customization** > **Property Mappings** and click **Create**. Create the following **SAML Provider Property Mapping**:
```

    - **Email mapping:**
        - **Name**: Choose a descriptive name.
        - **SAML Attribute Name**: `User.Email`
        - **Expression**:
            ```python
            return request.user.email
            ```

> 3. Optionally, create the following additional **SAML Provider Property Mapping** entries:

Reviewer: ```suggestion
3. Optionally, create the following additional **SAML Provider Property Mappings**:
```

1. Open the [Slack app creation page](https://api.slack.com/apps?new_app=1) and create an app for your workspace or organization.
2. In the app settings, click **OAuth & Permissions**.
3. Under **User Token Scopes**, add the `admin` scope.
4. Install the app to the workspace or organization that you want authentik to provision.
5. Copy the generated **User OAuth Token**. You need this token for the optional SCIM provider in authentik.

For Enterprise Grid, install the SCIM app on the organization, not on an individual workspace.

### Create a SCIM provider in authentik _(optional)_

Configure this optional provider only if you want authentik to provision Slack users and user groups through SCIM.

1. Return to the authentik Admin interface.
2. Navigate to **Applications** > **Providers** and click **Create**.
3. Select **SCIM Provider** as the provider type and click **Next**.
4. Configure the following settings:
    - **Provider Name**: enter a descriptive name.
    - **URL**: `https://api.slack.com/scim/v2/`
    - **Token**: paste the Slack user OAuth token.
    - **Compatibility Mode**: select **Slack**.
5. Click **Finish** to save the provider.

> 6. Navigate to **Applications** > **Applications** and open the Slack application that you created earlier.
> 7. Click **Edit** and add the SCIM provider to **Backchannel Providers**.
> 8. Click **Update**.
> 
> ## Configuration verification
> 

Reviewer: ```suggestion
8. Click **Update**.

Users that should be provisioned to Slack will then need to be bound to the Slack application. For more information on user and group filtering, refer to the [SCIM provider documentation](../../../providers/scim/#filter-the-sync-scope) .

## Configuration verification

```

To confirm that authentik is properly configured with Slack, open the integration from the authentik Application Dashboard. You should be redirected to Slack and signed in.

You can also open your Slack workspace directly and select the SSO option. You should be redirected to authentik to authenticate, then returned to Slack.

> If you configured SCIM provisioning, assign a test user or group to the Slack application in authentik and verify that the user or group is created or updated in Slack.

Reviewer: ```suggestion
If you configured SCIM provisioning, assign a test user to the Slack application in authentik and verify that the user is created or updated in Slack.
```
